contracts/strategies/donation-voting/DonationVotingStrategy.sol

```diff
@@ -0,0 +1,451 @@
+
+        address recipientAddress;
+        bool isRegistryIdentity;
+        Metadata memory metadata;
+
+        // decode data custom to this strategy
+        if (registryGating) {
+            (recipientId, recipientAddress, metadata) = abi.decode(_data, (address, address, Metadata));
+
+            if (!_isIdentityMember(recipientId, _sender)) {
+                revert UNAUTHORIZED();
+            }
+        } else {
+            (recipientAddress, isRegistryIdentity, metadata) = abi.decode(_data, (address, bool, Metadata));
+            recipientId = _sender;
+            if (isRegistryIdentity && !_isIdentityMember(recipientId, _sender)) {
+                revert UNAUTHORIZED();
+            }
+        }
+
+        if (recipientAddress == address(0)) {
+            revert INVALID();
+        }
+
+        Recipient storage recipient = _recipients[recipientId];
+
+        if (_recipients[recipientId].recipientAddress == address(0)) {
+            // new recipient
+            recipient.recipientAddress = recipientAddress;
+            recipient.isRegistryIdentity = registryGating ? true : isRegistryIdentity;
+            recipient.recipientStatus = InternalRecipientStatus.Pending;
+        } else {
+            // existing recipient
+            recipient.recipientAddress = recipientAddress;
+            if (recipient.recipientStatus == InternalRecipientStatus.Rejected) {
+                recipient.recipientStatus = InternalRecipientStatus.Appealed;
+                emit Appealed(recipientId, _data, _sender);
+            } else {
+                recipient.recipientStatus = InternalRecipientStatus.Pending;
+                emit Registered(recipientId, _data, _sender);
+            }
+        }
+    }
+
+    function _allocate(bytes memory _data, address _sender)
+        internal
+        virtual
+        override
+        nonReentrant
+        onlyActiveAllocation
+    {
+        (address recipientId, uint256 amount, address token) = abi.decode(_data, (address, uint256, address));
+
+        Recipient storage recipient = _recipients[recipientId];
+
+        if (!_isTokenAllowed(token) || recipient.recipientStatus != InternalRecipientStatus.Accepted) {
+            revert INVALID();
+        }
+
+        _transferAmount(token, address(this), amount);
+
+        claims[recipientId][token] += amount;
+
+        emit Allocated(recipientId, amount, token, _sender);
+    }
+
+    function _distribute(address[] memory __recipientIds, bytes memory, address _sender)
+        internal
+        virtual
+        override
+        onlyPoolManager(_sender)
+        onlyAfterAllocation
+    {
+        uint256 recipientLength = __recipientIds.length;
+        for (uint256 i; i < recipientLength;) {
+            address recipientId = __recipientIds[i];
+
+            Recipient storage recipient = _recipients[recipientId];
+
+            if (recipient.recipientStatus != InternalRecipientStatus.Accepted) {
+                revert INVALID();
+            }
```

Comment:
should we add the recipientId on revert?

Reply:
we should 🙌🏼